client/components/data/query-users-suggestions/index.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingUserSuggestions as isRequesting } from 'state/users/suggestions/selectors';
+import { requestUserSuggestions } from 'state/users/suggestions/actions';
+
+class QueryUsersSuggestions extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( props.isRequesting ) {
+			return;
+		}
+
+		props.requestUserSuggestions( props.siteId );
```

Comment:
@gwwar I'm not too sure of the circumstances under which that will occur, but regardless, I added a [check](https://goo.gl/wpHfvf) to ensure the suggestions aren't fetched in that scenario. Thx!